engine/crates/engine-v2/src/request/path.rs

```diff
@@ -1,78 +1,85 @@
-use schema::ObjectId;
+use schema::{ObjectId, Schema};
 
-use super::OperationFieldId;
-use crate::{
-    execution::StrId,
-    formatter::{ContextAwareDebug, FormatterContext, FormatterContextHolder},
-};
+use super::TypeCondition;
+use crate::execution::StrId;
 
 #[derive(Debug, Clone, Default, PartialEq, Eq)]
-pub struct OperationPath(im::Vector<OperationPathSegment>);
+pub struct QueryPath(im::Vector<QueryPathSegment>);
 
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct OperationPathSegment {
-    // Keeping the actual id around for debug/print/...
-    pub operation_field_id: OperationFieldId,
-    // Actual needed fields.
-    pub type_condition: Option<ResolvedTypeCondition>,
-    pub position: usize,
+pub struct QueryPathSegment {
+    pub resolved_type_condition: Option<ResolvedTypeCondition>,
     pub name: StrId,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct ResolvedTypeCondition(Vec<ObjectId>);
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ResolvedTypeCondition {
+    // sorted to guarantee deterministic order
+    possible_types: Vec<ObjectId>,
```

Comment:
Just to check my understanding: `...on MyObject` would be `vec![my_object_id]`,  `...on MyUnion` would be `union.possible_object_types().collect()`, and same on interface with the objects implementing the interface?

(all pseudocode ofc)

Reply:
Yeah exactly. The main purpose of `ResolvedTypeCondition` is to be used when finding the root response objects on which a plan needs to act. So supposing `catalog { products { price } }`, if `price` comes from a different subgraph we need to find all the `products` objects. They serve as an "anchor" to know where to add data later and also from where the input data must be read. Now as we could have nested type conditions, we resolve them eagerly to a list of possible object IDs.